lib/elixir/lib/partition_supervisor.ex

```diff
@@ -202,32 +202,63 @@ defmodule PartitionSupervisor do
   def start_child(mod, fun, args, name, partition) do
     case apply(mod, fun, args) do
       {:ok, pid} ->
-        :ets.insert(name, {partition, pid})
+        register_child(name, {partition, pid})
         {:ok, pid}
 
       {:ok, pid, info} ->
-        :ets.insert(name, {partition, pid})
+        register_child(name, {partition, pid})
         {:ok, pid, info}
 
       other ->
         other
     end
   end
 
+  defp register_child(name, key) when is_atom(name) do
+    :ets.insert(name, key)
+  end
+
+  defp register_child(name, key) when is_tuple(name) do
+    Registry.register(@registry, name, key)
```

Comment:
I actually think it is going to be more efficient to use `kind: :unique` and register the keys as:

    Registry.register(@registry, {parent_pid, partition}, [])

You don't need to store the pid, because that's the registry "key" anyway.